driver-core/src/test/functional/com/mongodb/client/model/expressions/DocumentExpressionsFunctionalTest.java

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model.expressions;
+
+import org.bson.BsonDocument;
+import org.bson.Document;
+import org.bson.conversions.Bson;
+import org.bson.types.Decimal128;
+import org.junit.jupiter.api.Test;
+
+import java.time.Instant;
+import java.util.Arrays;
+
+import static com.mongodb.client.model.expressions.Expressions.of;
+import static com.mongodb.client.model.expressions.Expressions.ofIntegerArray;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@SuppressWarnings("ConstantConditions")
+class DocumentExpressionsFunctionalTest extends AbstractExpressionsFunctionalTest {
+    // https://www.mongodb.com/docs/manual/reference/operator/aggregation/#object-expression-operators
+    // (Complete as of 6.0)
+
+    private static DocumentExpression ofDoc(final String ofDoc) {
+        return of(BsonDocument.parse(ofDoc));
+    }
+
+    private final DocumentExpression a1 = ofDoc("{a: 1}");
+    private final DocumentExpression ax1ay2 = ofDoc("{a: {x: 1, y: 2}}");
+
+    @Test
+    public void literalsTest() {
+        assertExpression(
+                BsonDocument.parse("{'a': 1}"),
+                ofDoc("{a: 1}"),
+                "{'$literal': {'a': 1}}");
+        assertThrows(IllegalArgumentException.class, () -> of((Bson) null));
+        // doc inside doc
+        assertExpression(
+                BsonDocument.parse("{'a': {'x': 1, 'y': 2}}"),
+                ofDoc("{a: {x: 1, y: 2}}"),
+                "{'$literal': {'a': {'x': 1, 'y': 2}}}");
+        // empty
+        assertExpression(
+                BsonDocument.parse("{}"),
+                ofDoc("{}"),
+                "{'$literal': {}}");
+        // ensure is literal
+        assertExpression(BsonDocument.parse(
+                "{'lit': {'$not': true}}"),
+                of(BsonDocument.parse("{lit: {'$not': true} }")),
+                "{'$literal': {'lit': {'$not': true}}}");
+    }
+
+    @Test
+    public void getFieldTest() {
```

Comment:
Please add tests for field names with embedded `.` characters so that we have regression tests on the behavior

Reply:
Also need to check if this is broken in the current implementation - on the first reading of the [getField](https://www.mongodb.com/docs/manual/reference/operator/aggregation/getField/) docs, it seemed it would be automatically treated as a literal, but I think it needs to be explicitly wrapped.

Reply:
Is this done?